kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/service/RuleCapabilitiesServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+ * limitations under the License.
+*/
+
+package org.kie.server.controller.impl.service;
+
+import java.util.List;
+
+import org.kie.server.api.model.KieScannerStatus;
+import org.kie.server.api.model.ReleaseId;
+import org.kie.server.controller.api.model.runtime.Container;
+import org.kie.server.controller.api.model.spec.Capability;
+import org.kie.server.controller.api.model.spec.ContainerConfig;
+import org.kie.server.controller.api.model.spec.ContainerSpecKey;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.api.service.NotificationService;
+import org.kie.server.controller.api.service.RuleCapabilitiesService;
+import org.kie.server.controller.api.storage.KieServerTemplateStorage;
+import org.kie.server.controller.impl.KieServerInstanceManager;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.RuleConfig;
+import org.kie.server.controller.impl.storage.InMemoryKieServerTemplateStorage;
+
+public class RuleCapabilitiesServiceImpl implements RuleCapabilitiesService {
+
+    private KieServerTemplateStorage templateStorage = InMemoryKieServerTemplateStorage.getInstance();;
+    private KieServerInstanceManager kieServerInstanceManager = KieServerInstanceManager.getInstance();
+    private NotificationService notificationService = LoggingNotificationService.getInstance();
+
+    @Override
+    public void scanNow(final ContainerSpecKey containerSpecKey) {
+        ServerTemplate serverTemplate = templateStorage.load(containerSpecKey.getServerTemplateKey().getId());
+        if (serverTemplate == null) {
+            throw new RuntimeException("No server template found for id " + containerSpecKey.getServerTemplateKey().getId());
+        }
+
+        org.kie.server.controller.api.model.spec.ContainerSpec containerSpec = serverTemplate.getContainerSpec(containerSpecKey.getId());
+
+        List<Container> containers = kieServerInstanceManager.scanNow(serverTemplate, containerSpec);
+
+        notificationService.notify(serverTemplate, containerSpec, containers);
+    }
+
+    @Override
+    public void startScanner(final ContainerSpecKey containerSpecKey, long interval) {
+
+        ServerTemplate serverTemplate = templateStorage.load(containerSpecKey.getServerTemplateKey().getId());
+        if (serverTemplate == null) {
+            throw new RuntimeException("No server template found for id " + containerSpecKey.getServerTemplateKey().getId());
+        }
+
+        org.kie.server.controller.api.model.spec.ContainerSpec containerSpec = serverTemplate.getContainerSpec(containerSpecKey.getId());
+
+        ContainerConfig containerConfig = containerSpec.getConfigs().get(Capability.RULE);
+        if (containerConfig == null) {
+            containerConfig = new RuleConfig();
+            containerSpec.getConfigs().put(Capability.RULE, containerConfig);
+        }
+
+        ((RuleConfig) containerConfig).setPollInterval(interval);
```

Comment:
should be, otherwise it must fail, config under Capability.RULE must be RuleConfig